### PR TITLE
[FSSDK-12647] user id vuid adjustment

### PR DESCRIPTION
**Pull request by junaed-optimizely**
## Summary
- Add null support to the Provider's user prop type
- VUID mode now requires an explicit empty object (user={{}}) instead of omitting user
## Test plan
Tests have been added to support the change

## Issues
- FSSDK-12647

### CODE REVIEW

File: src/provider/OptimizelyProvider.spec.tsx [hunk 1]
```diff
@@ -793,6 +793,56 @@ describe('OptimizelyProvider', () => {
...
+      expect(mockClient.createUserContext).not.toHaveBeenCalled();
+      expect(capturedContext!.store.getState().userContext).toBeNull();
+
+      rerender(
+        <OptimizelyProvider client={mockClient} user={{ id: 'user-1' }}>
+          <ContextConsumer onContext={(ctx) => (capturedContext = ctx)} />
+        </OptimizelyProvider>
+      );
+
+      expect(mockClient.createUserContext).toHaveBeenCalledWith('user-1', undefined);
```
Comment (raju-opti) [inline src/provider/OptimizelyProvider.spec.tsx:842]: can we also add an assertion that the userContext is present in the store?

File: src/provider/OptimizelyProvider.spec.tsx [hunk 2]
```diff
@@ -793,6 +793,56 @@ describe('OptimizelyProvider', () => {
...
+    });
+  });
+
```
Comment (raju-opti) [inline src/provider/OptimizelyProvider.spec.tsx:867]: can we also add a test for when user prop goes from valid to null that the store userCOntext also set to null?

**Review by raju-opti — APPROVED** [review]
